Initial Commit

ytv opened the pull request:
Initial implementation of mockDB tool includes config, package.JSON, and helper scripts: crud-service, doc-config and usage.

The only crud services supported so far are view, insert, and delete (all).  To display usage, type in "node mockDB.js"

I want to able to access the other crud services (find, update, replace) from the command line but to do so, I need to be able to pass in 2 different JSON objects as command line arguments.  For example, in order to update a document (or records, as they call it in SQL), I need to somehow pass in {field1: value1} and {field2: value2} so that it can find the documents whose field1 has value1 and update field2 with value 2.

I was able to google libraries like "minimalist" and "yargs" that parse out command line key-value arguments, but they don't seem to allow me to separate key-value arguments into 2 categories: key-value pairs I'm using as criteria to find documents and key-value pairs I'm using to update the documents with.  Does this make sense, and if so, any ideas?

ytv committed:
Initial Commit

Initial implementation of mockDB tool includes config, package.JSON, and helper scripts: crud-service, doc-config and usage.

The only crud services supported so far are view, insert, and delete (all).

darrenvng commented:
Hmm ... I'm not sure what you mean, but the advice I have been given which works very well is: '**If things start getting too complicated, its a sign your going in the wrong direction**'.   Start simply with what we discussed and build one piece, one module at a time.  All your version 1 needs to do is 'node <my-script.js> users <count>'.  Your optimizations can be version 2??  Always make sure you accomplish your initial tasks before expanding into other areas, open a github issue if you need to track pending optimzations/features/bugs.

Proper way to address your problem is, I want to do something like this 'node <my-script.js> type=user size=100' and have my script utilize these user params.  How would I go about doing this?  Don't start with technical specifics until prompted by the person you are asking since they might not understand what you are trying to achieve!

darrenvng commented on mockDB.js:
```diff
@@ -0,0 +1,103 @@
+var config = require('./config.js'),
```
follow the proper node js naming conventions for files, mock-db.js?

ytv replied:
Changed to instant-db.js

darrenvng commented on config.js:
```diff
@@ -0,0 +1,8 @@
+var config = {};
+
```
module.exports = {
  db: '', //type 'mongo'
  url: '' 
};

This is designed for user customization!

ytv replied:
Changed

darrenvng commented on config.js:
```diff
@@ -0,0 +1,8 @@
+
+config.db.url = 'mongodb://localhost/test';
+// config.db.url = 'mongodb://admin:password@ds017248.mlab.com:17248/mockdb';
+
```
user has to type a lot of extra letters rather than exactly what they need.

ytv replied:
Changed

darrenvng commented on mockDB.js:
```diff
@@ -0,0 +1,103 @@
+    crudService = require('./scripts/crud-service.js'),
+    doc = require('./scripts/doc-config.js'),
+    usage = require('./scripts/usage.js');
+
```
Use config folder, so you don't have one giant config object.  config.db.mongo.url.... is a bad sign.  Should just be an object 1 level deep.

ytv replied:
Done

darrenvng commented on mockDB.js:
```diff
@@ -0,0 +1,103 @@
+    usage = require('./scripts/usage.js');
+
+var url = config.db.url,
+    cmds = [
```
Use a map! This is an O(N) implementation when it can be O(1)!  Also, put this in a separate config file if you need to access in other parts of your app.

ytv replied:
Done

darrenvng commented on mockDB.js:
```diff
@@ -0,0 +1,103 @@
+            db.insertMany(collection, args[2], getData());
+        }
+    }
+    else if (command === 'delete') {
```
command types shouldn't be hard coded, use a mapped config and check against it.

ytv replied:
Done

darrenvng commented on scripts/crud-service.js:
```diff
@@ -0,0 +1,87 @@
+        mongo.connect(url, function(err, db) {
+            if(err) throw "ERROR connecting to DB - " + err;
+            db.collection(collectionName).find(query).toArray(function(err, docs) {
+                if(err)
```
nice, success / error handling lets users know whats going on.

darrenvng commented on scripts/doc-config.js:
```diff
@@ -0,0 +1,26 @@
+
+module.exports = {
+    generateUser: function() {
+        return {
```
Definitely need images!

ytv replied:
Added

darrenvng commented on mockDB.js:
```diff
@@ -0,0 +1,103 @@
+    should be in JSON format */
+var db = {
+
+    find: function(collection, query) {
```
These methods belong in your CMS no?  This app should only focus on generating / deleting content from user DB.  I  think we need to recap what the goals of each of the 3 projects should be.  Crud services are for communication with a client device!!  Does not belong in a node script!

ytv replied:
Removed

ytv committed:
Initial Commit
Moved config files into one folder

ytv committed:
Initial Commit

Renamed from mockDB.js

ytv committed:
Initial Commit

Updated usage to allow for any collection name and removed usage regarding find/update/replacement

ytv committed:
Initial Commit

Simplified services by removing replacement/find/update

ytv committed:
Initial Commit

Revised README to include usage and notes

ytv committed:
Initial Commit

ytv commented:
Is this good to merge with the new changes?
